token/program-2022/src/instruction.rs

```diff
@@ -692,11 +836,45 @@ impl AuthorityType {
             1 => Ok(AuthorityType::FreezeAccount),
             2 => Ok(AuthorityType::AccountOwner),
             3 => Ok(AuthorityType::CloseAccount),
+            4 => Ok(AuthorityType::FeeConfig),
+            5 => Ok(AuthorityType::FeeWithdraw),
             _ => Err(TokenError::InvalidInstruction.into()),
         }
     }
 }
 
+/// Specifies the authority type for SetAuthority instructions
+#[derive(Clone, Debug, PartialEq)]
+pub enum InitializeMintMixinParameters {
+    /// Additional padding required, in case the size of the account is the same
+    /// as a multisig
+    None {
+        padding: u64,
+    },
+    /// Initializes a mint with the `MintTransferFeeMixin`
+    TransferFee {
+        /// Pubkey that may update the fees
+        fee_config_authority: COption<Pubkey>,
+        /// Harvest instructions must be signed by this key
+        fee_withdraw_authority: COption<Pubkey>,
+        /// Amount of transfer collected as fees, expressed as basis points of the
+        /// transfer amount
+        transfer_fee_basis_points: u16,
+        /// Maximum fee assessed on transfers
+        maximum_fee: u64,
+    },
+}
```

Comment:
Just for context, I immediately see a need to add a Mint-level mixin:
   `CloseAccount { close_authory:  COption<Pubkey> }`,
which will permit a `Mint` account to be closed via `CloseAccount` when the supply is 0.  This can be used to reclaim future NFT mints (cc: @bartosz-lipinski)

Reply:
I don't think so, seems like it's just a mint-level setting